### PR TITLE
PublicDashboards: Remove unsupported datasource warning for cloudwatch

### LINKED ISSUE #75243: Remove unsupported DS map and update documentation
Remove [unsupported DS map ](https://github.com/grafana/grafana/blob/main/pkg/services/publicdashboards/commands/generate_datasources/generate.go#L16) once [this task](https://github.com/grafana/grafana/issues/74872) is done

### PR DESCRIPTION
**Pull request by juanicabanas**
**What is this feature?**

After [this PR](https://github.com/grafana/grafana/pull/77532), cloudwatch is extending from `DataSourceWithBackend` and executing `super.query`. That allows public dashboard code to be executed and to work properly.
This PR removes the warning in the public dashboard modal and cloudwatch from the unsupported data sources list in the documentation.

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/75243

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/dashboards/dashboard-public/index.md
Comment (AgnesToulet) [inline docs/sources/dashboards/dashboard-public/index.md:1]: I think this should not only be removed from the "Unsupported" box but added to the "Confirmed" box above.

File: pkg/services/publicdashboards/commands/generate_datasources/generate.go
Comment (AgnesToulet) [inline pkg/services/publicdashboards/commands/generate_datasources/generate.go:1]: As I don't think Cloudwatch was the only unsupported datasource, I think we should keep the variable and the check but just use an empty map for now (with maybe an updated comment saying that we still need to go through the datasources we may not support before getting rid of this). I'm thinking at Splunk that we may need to add here in the near future.

**Review by AgnesToulet — APPROVED** [review]
LGTM but I left a couple of suggestions.

File: docs/sources/dashboards/dashboard-public/index.md
Comment (juanicabanas) [reply docs/sources/dashboards/dashboard-public/index.md:1]: sorry, I missed it. I agree

File: pkg/services/publicdashboards/commands/generate_datasources/generate.go [hunk 1]
```diff
@@ -45,7 +40,7 @@ func GetCompatibleDatasources(baseUrl string) ([]string, error) {
 	// we only consider a datasource to be supported when alerting and backend are both true
 	var supported []string
 	for _, datasource := range datasources {
-		if datasource.Alerting && datasource.Backend && !unsupportedDataSourcesMap[datasource.Slug] {
+		if datasource.Alerting && datasource.Backend {
```
Comment (evictorero) [inline pkg/services/publicdashboards/commands/generate_datasources/generate.go:43]: I would hold this change until we have a confirmation about Splunk. Right now, it should be on the unsupported Map.

**Pull request merged**